<?xml version="1.0"?>
<launch>
  <arg name="joy_dev" default="/dev/input/js0" />
  <arg name="ip" default="169.254.148.86" />
  <arg name="port" default="3883" />

  <!-- for CF num 1
  <arg name="frame" default="Crazyflie_1" />
  <arg name="uri" default="radio://0/80/250K" />
  -->

  <!-- for CF num 3   -->
  <arg name="frame" default="Crazyflie_5" />
  <arg name="uri" default="radio://0/80/250K/E7E7E7E7E7" />

  <!-- for CF num 5
  <arg name="frame" default="Crazyflie_5" />
  <arg name="uri" default="radio://0/80/250K/E7E7E7E7E9" />
  -->


  <!-- crazyflie name should be the same as labeld in OPTITRACK -->

  <arg name="worldFrame" default="crazyflie/base_link" />

  <!-- Pose -->
  <arg name="x" default="0" />
  <arg name="y" default="0" />
  <arg name="z" default="1.0" />


  <include file="$(find crazyflie_driver)/launch/crazyflie_server.launch">
  </include>

  <group ns="crazyflie">
    <include file="$(find crazyflie_driver)/launch/crazyflie_add.launch">
      <arg name="uri" value="$(arg uri)" />
      <arg name="tf_prefix" value="crazyflie" />


      <!-- Logging is disabled to reduce unnecessary bandwith consumption-->
      <arg name="enable_logging" value="True" />
      <param name = "enable_logging_imu" value = "True" />
      <param name = "enable_logging_temperature" value = "False" />
      <param name = "enable_logging_magnetic_field" value = "False" />
      <param name = "enable_logging_pressure" value = "False" />
      <param name = "enable_logging_battery" value = "False" />
      <param name = "enable_logging_packets" value = "False" />

    </include>


      <include file="$(find crazyflie_controller)/launch/crazyflie2.launch">
        <arg name="frame" value="$(arg frame)" />
      </include>

      <!-- launch pose_publisher with given parameter -->
      <node name = "pose" pkg = "crazyflie_demo" type = "publish_pose.py" output = "screen" >
        <param name = "worldFrame" value = "crazyflie/base_link" />
        <param name = "name" value = "goal" />
        <param name = "rate" value = "30" />
        <param name = "x" value = "$(arg x)" />
        <param name = "y" value = "$(arg y)" />
        <param name = "z" value = "$(arg z)" />
      </node>

      <node pkg="tf" type="static_transform_publisher" name="baselink_broadcaster" args="0 0 0 0 0 0 1 $(arg frame) /crazyflie/base_link 100" />

    <!-- run vrpn client
      <node pkg="vrpn_client_ros" name="vrpn" output="screen" type="vrpn_client_node" required="true">
        <rosparam subst_value="true">
          server: $(arg ip)
          port: $(arg port)
          update_frequency: 100.0
          frame_id: /world
          child_frame_id: $(arg frame)
          use_server_time: true
          broadcast_tf: true
          refresh_tracker_frequency: 1.0
        </rosparam>
      </node>
    -->

  </group>


  <!-- Create a group for open_loop_controller  -->
  <group ns="open_loop_controller">
    <node name="keyboard_controller" pkg="crazyflie_demo" type="keyboard_controller.py" output="screen" launch-prefix="xterm -e" >
      </node>
  </group>


  <!-- run rviz -->
  <param name="robot_description" command="$(find xacro)/xacro.py $(find crazyflie_description)/urdf/crazyflie.urdf.xacro" />
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find crazyflie_demo)/launch/crazyflie_pos.rviz" required="true" />

</launch>
